src/entity.ts

```diff
@@ -271,6 +271,9 @@ export class Entity {
 					// Remove excess items from the list
 					currentValue.splice(state.length);
 				}
+
+				// Mark property pendingInit = false, since the setter is not called
```

Comment:
Its saying that if the setter was called it would set pending init to false, but since its not called we have to set pending init here.

Reply:
If the `value` variable is set, then it calls the setter at the bottom of this method, and the setter will set pending init = false. In some cases value is not assigned and instead the current value (i.e. array) is directly manipulated. If this resulted in changes to the array then the array change handler would set pending init. But, if there were no changes to the array then it wouldn't, so that's why I set it here as well.

Reply:
Honestly it probably _should_ just call the setter. But, I'd rather not pull that into the scope of this PR. 😬

Reply:
Yeah, I think that's the idea in theory, that its going to take the given raw state and set the property. In many cases it does just that (calls the setter), but list properties are just a bit of an edge case since it directly modifies the list instead. It may be that it was done this way because until recently the setter would throw an error if you called it for a list property.